src/server/pfs/server/driver.go

```diff
@@ -266,9 +266,25 @@ func (d *driver) getPermissions(ctx context.Context, repo *pfs.Repo) ([]auth.Per
 	return resp.Permissions, resp.Roles, nil
 }
 
-func (d *driver) listRepo(ctx context.Context, includeAuth bool, repoType string, projects []*pfs.Project, cb func(*pfs.RepoInfo) error) error {
-	authSeemsActive := true
-	repoInfo := &pfs.RepoInfo{}
+func (d *driver) getPermissionsInTransaction(txnCtx *txncontext.TransactionContext, repo *pfs.Repo) ([]auth.Permission, []string, error) {
+	resp, err := d.env.AuthServer.GetPermissionsInTransaction(txnCtx, &auth.GetPermissionsRequest{Resource: repo.AuthResource()})
+	if err != nil {
+		return nil, nil, errors.EnsureStack(err)
+	}
+
+	return resp.Permissions, resp.Roles, nil
+}
+
+// NOTE if includeAuth = true, it is expected that auth is active
+func (d *driver) processListRepoInfo(
```

Comment:
Why did we choose to model some of the logic as function parameters instead of putting that logic inside this method itself? Is that for unit testing purposes?